### ISSUE TITLE
Suggestions for release installation?

### ISSUE THREAD

**Issue by ablakey** [OSHistory/generate_ol_types#1]
Thanks for making this. It's a great stop-gap until Openlayers types are available sometime in February or later.

Do you have any suggestions on the best way to install the release typings? I manually copied them into node_modules/@types/ but that's generally not a sustainable strategy.

I wonder if they can be made available on npm or in a repo that can be included as a package.json dependency. Then there may be a clean way in TypeScript to include that source for types as well?

I'm still learning TS but I'd be happy to help in whatever way possible if you think there's value in doing so.

**Comment by OSHistory** [comment]
Yes, you're right. The process is less than satisfactory. But I didn't want to publish them 
officially over npm / DefinetlyTyped until some rudimentary quality level is reached. For 
the moment a lot of the type information in the ol-source is getting lost. The greatest 
benefit thus far is a better development experience by providing the documentation and code completion without tweaks to the setup. 

I think the idea of separately publishing to a github repository would be the most feasible, 
since the naming could make clear, that the typings are in beta state.

Regarding the further development:
I have been investigating to alternative approaches, but they haven't been very successful 
yet :( 

One has been a manual approach to the ol-source parsing (thus not relying on the current 
hacky setup of forcing the js-source to ts to generate the typings). While the parser worked 
quite well, the real problem comes with the dependency management (which is not figured out 
by Typescript, but has to be done manually).

The other bases on a suggestion from this issue: 
https://github.com/openlayers/openlayers/issues/8448#issuecomment-417052111
I have a branch on my OpenLayers Fork that passes the typecheck without 
errors (https://github.com/OSHistory/openlayers/tree/fixRemainingTsc)
and thus can generate valid TypeDoc json. This seems like the more 
promising way to go. I'll create another repo once I have some basic 
setup.

**Comment by OSHistory** [comment]
@nipuna777 
As an update to the version6 release checkout this recently posted issue. 

https://github.com/openlayers/openlayers/issues/9191

**Comment by OSHistory** [comment]
@ablakey @nipuna777 
I started an issue to keep track of development to better TS-support.
https://github.com/OSHistory/generate_ol_types/issues/2

**Comment by OSHistory** [comment]
@wulfsberg 
Picking up on your comment in the OL-thread 
(https://github.com/openlayers/openlayers/pull/9178#issuecomment-461724167).  
I pinged you to make you aware of this humble effort 
of mine, which is currently a bit messy. Generally speaking, automatic  generation 
of .d.ts files is quite tricky.  

In regards to the suggested packages, I think I remember testing either one and failing. If you
do find the time to investigate either one, I'd be gratefull if you could share your insights here.

**Comment by wulfsberg** [comment]
Yeah, the type generation I have looked at so far has been pretty brittle. My hope is that since the OL JsDoc is specifically written to be TypeScript compatible, it'll be easier to process correctly, but I have yet to try anything with the new OL 6 source.
I hope to do some test runs, but I have never really used the JsDoc-for-TypeScript, so I am feeling my way around.

Somewhat tangentially, it feels like we're weirdly somewhere midstream, with the JsDoc types *almost* working, and TypeScript understanding the typings, but somehow forgetting them in the middle of the build step. It seems strange that TypeScript can use them for full checking, but not that last step of writing out the typings it already used internally, so I am holding out some hope that this could be fixed for good with https://github.com/Microsoft/TypeScript/issues/7546

**Comment by OSHistory** [comment]
Yes indeed, this would be our *deus ex machina* :)

**Comment by OSHistory** [comment]
A new effort has come up using a jsdoc plugin:
https://github.com/openlayers/openlayers/pull/8345#issuecomment-486895571

**Comment by OSHistory** [comment]
As I learned from an issue the resulting typings have been merged to definetly typed (`npm install @types/ol`). They do seem to 
work very well. I suggest following this approach, as it is more concise. I will mark this project as 
deprecated.
